### ISSUE TITLE
msgpack: test_phil.py failing

### ISSUE THREAD

**Issue by rjgildea** [dials/dials#762]
The following test is currently failing, I believe it is related to the msgpack changes:
`$ pytest --regression --runslow -n auto test/test_phil.py`

```
test/test_phil.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../cctbx_project/libtbx/phil/__init__.py:1780: in extract
    value = object.extract(parent=result)
../cctbx_project/libtbx/phil/__init__.py:1780: in extract
    value = object.extract(parent=result)
../cctbx_project/libtbx/phil/__init__.py:1144: in extract
    return self._type_from_words()(self.words, master=self)
util/phil.py:90: in from_words
    return self.from_string(libtbx.phil.str_from_words(words=words))
util/phil.py:86: in from_string
    self.cache[s] = FilenameDataWrapper(s, flex.reflection_table.from_file(s))
array_family/flex.py:286: in from_file
    return reflection_table.from_pickle(filename)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

filename = '/dls/science/groups/scisoft/DIALS/dials_data/centroid_test_data/integrated.mpack'

    @staticmethod
    def from_pickle(filename):
        """
        Read the reflection table from pickle file.
    
        :param filename: The pickle filename
        :return: The reflection table
    
        """
        import six.moves.cPickle as pickle
        from libtbx import smart_open
    
        if filename and hasattr(filename, "__fspath__"):
            filename = filename.__fspath__()
        with smart_open.for_reading(filename, "rb") as infile:
>           result = pickle.load(infile)
E           UnpicklingError: invalid load key, ''.

array_family/flex.py:225: UnpicklingError
```

```
$ dials.python
Python 2.7.16 (default, Apr  3 2019, 15:01:33) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-23)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from dials.array_family import flex
>>> refl = flex.reflection_table.from_file("/dls/science/groups/scisoft/DIALS/dials_data/centroid_test_data/integrated.mpack")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/dls/mx-scratch/rjgildea/software/cctbx/modules/dials/array_family/flex.py", line 286, in from_file
    return reflection_table.from_pickle(filename)
  File "/dls/mx-scratch/rjgildea/software/cctbx/modules/dials/array_family/flex.py", line 225, in from_pickle
    result = pickle.load(infile)
cPickle.UnpicklingError: invalid load key, ''.
>>> refl = flex.reflection_table.from_msgpack("/dls/science/groups/scisoft/DIALS/dials_data/centroid_test_data/integrated.mpack")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: dials Error: /dls/mx-scratch/rjgildea/software/cctbx/modules/dials/array_family/reflection_table_msgpack_adapter.h(692): msgpack type is not an array
```

**Comment by jmp1985** [comment]
Fixed in 6e09482749eee5b34c5674e667f824080d8969d9

**Comment by ndevenish** [comment]
To continue @Anthchirp's comment; this is basically dead code that will never hit because we only just switched and it's failing on a "pre-mpack" messagepack file?

Wouldn't it be better to just update this and not have this test path in permanently? Or at least a comment saying why this test is sometimes required...
